### PR TITLE
Add `rfm_train_test_split` Utility Function

### PR DESCRIPTION
**Pull request by ColtAllen**
## Description

This PR adapts the `calibration_and_holdout_data` function from `lifetimes` to evaluate model performance for MAP fits, and can also be used to estimate the impact of a causal intervention like a marketing campaign:

image

This PR also fixes [a known bug](https://github.com/CamDavidsonPilon/lifetimes/issues/431) in the `lifetimes` function.

I've added a dev notebook to test this and other CLV utilities, but it may be removed after this functionality is fully built out.

It's important to note this function will exclude new customers whose first transaction occurred during the test period. I'm torn on what to do about this - should a separate, additional `rfm_summary` be returned for these new customers? Or just not worry about it?

## Checklist

- [x] Checked that [the pre-commit linting/style checks pass](https://docs.pymc.io/en/latest/contributing/python_style.html)
- [x] Included tests that prove the fix is effective or that the new feature works
- [x] Added necessary documentation (docstrings and/or example notebooks)
- [x] If you are a pro: each commit corresponds to a [relevant logical change](https://wiki.openstack.org/wiki/GitCommitMessages#Structural_split_of_changes)

## Modules affected

- [ ] MMM
- [x] CLV

## Type of change

- [x] New feature / enhancement
- [ ] Bug fix
- [ ] Documentation
- [ ] Maintenance
- [ ] Other (please specify):

----
📚 Documentation preview 📚: https://pymc-marketing--572.org.readthedocs.build/en/572/

### CODE REVIEW

**Comment by juanitorduz** [comment]
Very cool! I left a comment about making the code more modular. 

Also, an example notebook would be very helpful! You can either do it on this PR or create an issue and tackle it on a separate one 💪

File: pymc_marketing/clv/utils.py [hunk 1]
```diff
@@ -410,3 +411,171 @@ def rfm_summary(
     )
 
     return summary_df
+
+
+def rfm_train_test_split(
```
[the remark this answers is not in the document]
Comment (juanitorduz) [reply pymc_marketing/clv/utils.py:416]: Also need to add `rfm_train_test_split` to `__all__`

File: pymc_marketing/clv/utils.py [hunk 2]
```diff
@@ -410,3 +411,171 @@ def rfm_summary(
...
+            right_on=customer_id_col,
+            how="inner",
+        )
+        test_rfm_data = test_rfm_data.rename(
+            columns={monetary_value_col: "test_monetary_value"}
+        )
+    train_test_rfm_data = training_rfm_data.merge(
+        test_rfm_data, on="customer_id", how="left"
+    )
+    # TODO: What to do with new customers not in the train period? Disregard, or return in a second DF?
```
[the remark this answers is not in the document]
Comment (ColtAllen) [reply pymc_marketing/clv/utils.py:573]: I've added a comment about it in the docstrings.

**Comment by ColtAllen** [comment]
> Very cool! I left a comment about making the code more modular.

Could you be more specific? Two other utility functions are already being called internally by this method for the reusable bits.

> Also, an example notebook would be very helpful! You can either do it on this PR or create an issue and tackle it on a separate one 💪

The dev notebook in this PR could be adequate. For an example in a more comprehensive tutorial I'd prefer a separate PR because this method has a companion plotting function in `lifetimes` that would be good to add first.

**Review by ricardoV94 — APPROVED** [review]

**Pull request merged**